### DISCUSSION TITLE
Most commonly uninstalled, safe to uninstall and bloatware removal

### DISCUSSION THREAD

**Discussion by TCMOREIRA** [Klocman/Bulk-Crap-Uninstaller#530] [Ideas]
I love this program, I use it all the time. It's near perfect. There's only two missing capabilities in my opinion. 

1) My usage of BCU too often goes as follows: I have to web-search each program, one by one, to make sure that's safe to uninstall it and that that program is often uninstalled by everyone. It's nice that BCU allows to right-click on the application to web-search it, but I wish this process was shortened by BCU having data of which applications are safe and most commonly uninstalled, as rated by both its community of users and speciliazed websites. Thus allowing the user to filter those applications.

2) I wish I could use BCU as a bloatware removal tool. There are often way too many programs that come pre-installed with our computers that are actually useless. To be able to easily remove those using BCU would mean not having to download another uninstaller program that focuses only on bloatware removal. I often end up having to run both some bloatware removal tool followed by BCU, I'd love to optimize that pipeline.

Some kinds of Bloatware stick around and are undetected by all major bloatware removal tools (believe me I tried at least a dozen of them). That's where BCU always comes in handy! It shows me everything that's installed on the computer even after an automatic bloat-wipe. However again, this process is tedious and sometimes dangerous. I'd love to be able to automatically filter all of the remaining bloat without having to web-search each program one-by-one to see what it does. The culprits are usually manufacturer programs (DELL, Lenovo, Acer, ASUS, HP, Alienware, Razer, etc), garbage middleware applications and other kinds of junk such as toolbars that get installed when installing a printer's drivers.

**Comment by popy2k14** [comment]
Also love BCU!

Another idea in this direction is that users have a centralized place to share their uninstall lists and rate them (maybe in another official BCU git repo). This could be managed by PR's.
BCU can show/use those lists.
This way, everyone can share his filter list with others.

My idea are default list's like:

- MS bloatware
- ...

**Reply by Klocman** [reply]
I like the idea of having a repo with an uninstall list, it seems like the simplest and most maintainable option. Anyone who's up to the task can make a repo like this too.

**Reply by TCMOREIRA** [reply]
There already exist two related projects that do maitain such lists; in a way that I believe would greatly accelerate such endeavour; those being:

- [BloatyNosy](https://github.com/builtbybel/BloatyNosy)

    _And_

- [Bloatbox](https://github.com/builtbybel/bloatbox)

**Reply by Klocman** [reply]
Both of these look like the same project underneath, they both appear to be using the same app list. It's only listing official Microsoft apps to remove, so it's very limited in usability and more of a privacy tool than debloat.

**Reply by TCMOREIRA** [reply]
Oh! Sorry. Yes. They're from the same author. One is just an extension of the other.

Didn't mean to imply this would solve all problems or anything like that.
Just that that was a good place to start looking and that I remembered they already had a similar list-keeping concept; even if their list is focused on microsoft apps. [However, from what I remember, they do support addons where you can download other people's lists and such.]

I know there are other such projects, so I imagine taking a look at their lists would be a good idea.

**Reply by Klocman** [reply]
Yeah, it's a good start if anyone wanted to make a repo specifically for BCU uninstall lists.
